### PR TITLE
add mlockall and munlockall

### LINKED ISSUE #875: mlockall/munlockall
It would be great if nix had mlockall/munlockall in addition to mlock/munlock.

I'd be happy to try and add it. Or is there a specific reason why it was omitted?

### PR DESCRIPTION
**Pull request by afck**
Closes #875

### CODE REVIEW

File: src/sys/mman.rs [hunk 1]
```diff
@@ -213,6 +223,14 @@ pub unsafe fn munlock(addr: *const c_void, length: size_t) -> Result<()> {
     Errno::result(libc::munlock(addr, length)).map(drop)
 }
 
+pub unsafe fn mlockall(flags: MlockAllFlags) -> Result<()> {
```
Comment (asomers) [inline src/sys/mman.rs]: Why are these functions `unsafe`?

Comment (afck) [reply src/sys/mman.rs]: You're right, I suppose there's nothing actually unsafe here. Done.

**Comment by afck** [comment]
I conditionalized the constants for now.

**Comment by asomers** [comment]
Could you please check Linux/S390x?  I'd rather not needlessly disable obscure platforms simply out of neglect.

**Comment by afck** [comment]
I'm not sure how to verify that; it looks like the constants are defined in `tools/include/uapi/asm-generic/mman.h`, which is included in `tools/arch/s390/include/uapi/asm/mman.h`, so I guess it is? I'll have a look at the libc crate.

**Comment by asomers** [comment]
libc merged the PR, so you should be able to update this PR as well.  No need to update Nix's Cargo.toml.

**Comment by afck** [comment]
Thank you, I removed the conditions.

**Comment by asomers** [comment]
Could you please add some doc comments for the new functions, too?

**Comment by afck** [comment]
I also documented `mlock`/`munlock`. Please check whether you're okay with my formulation.

**Comment by asomers** [comment]
Ok, I think we're done here.

bors r+

**Comment by asomers** [comment]
Hm, bors's behavior has changed.  It appears that I now need to click "approve changes" _before_ telling bors to merge.

bors r+

**Comment by asomers** [comment]
Let's retry.

bors r+

**Pull request merged**